### PR TITLE
Add note explaining MLflow Evaluate precision_recall_auc metric calculation

### PR DESCRIPTION
**Pull request by BenWilson2**
Signed-off-by: Ben Wilson <benjamin.wilson@databricks.com>

## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Adds an explanation of how the unified metric `precision_recall_auc` is derived from two separate algorithms based on whether the classifier under evaluation is multi-class or binary. 

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests (describe details, including test results, below)

Generated DOCS locally

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [X] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/models.rst [hunk 1]
```diff
@@ -1634,6 +1634,29 @@ been logged: 'baseline_model' and 'candidate_model' for comparison purposes in t
 Additional information about model evaluation behaviors and outputs is available in the
 :py:func:`mlflow.evaluate()` API docs.
 
+.. note:: Differences in the computation of Area under Curve Precision Recall score (metric name
+    ``precision_recall_auc``) between multi and binary classifiers:
+
+    Multiclass classifier models, when evaluated, utilize the standard scoring metric from sklearn:
+    ``sk_metrics.roc_auc_score`` to calculate the area under the precision recall curve. This
```
Comment (harupy) [inline docs/source/models.rst:1641]:
```suggestion
    ``sklearn.metrics.roc_auc_score`` to calculate the area under the precision recall curve. This
```

`sk_metrics` sounds like a third-party packages that provides functions to compute metrics.

File: docs/source/models.rst [hunk 2]
```diff
@@ -1634,6 +1634,29 @@ been logged: 'baseline_model' and 'candidate_model' for comparison purposes in t
...
+    algorithm performs a linear interpolation calculation utilizing the trapezoidal rule to estimate
+    the area under the precision recall curve. It is well-suited for use in evaluating multi-class
+    classification models to provide a single numeric value of the quality of fit.
+
+    Binary classifier models, on the other hand, use the ``sk_metrics.average_precision_score`` to
```
Comment (harupy) [inline docs/source/models.rst:1646]:
```suggestion
    Binary classifier models, on the other hand, use the ``sklearn.metrics.average_precision_score`` to
```

**Review by harupy — APPROVED** [review]
Left comments, otherwise LGTM!

File: docs/source/models.rst [hunk 3]
```diff
@@ -1634,6 +1634,29 @@ been logged: 'baseline_model' and 'candidate_model' for comparison purposes in t
...
+    avoid the shortcomings of the ``roc_auc_score`` implementation when applied to heavily
+    imbalanced classes in binary classification. Usage of the ``roc_auc_score`` for imbalanced
+    datasets can give a misleading result (optimistically better than the model's actual ability
+    to accurately predict the minority class membership).
+
+    For additional information on the topic of why different algorithms are employed for this, as
+    well as links to the papers that informed the implementation of these metrics within the
+    ``sk_metrics`` library, refer to
```
Comment (harupy) [inline docs/source/models.rst:1654]:
```suggestion
    ``sklearn.metrics`` module, refer to
```
